### ISSUE TITLE
Rename de game.whenCollideDo

### ISSUE THREAD

**Issue by asanzo** [uqbar-project/wollok#1818]
Me parece importante tener una buena separación en el **nombre** y también en el "javadoc" entre `onCollideDo` y el actual `whenCollideDo`. Hoy son casi idénticas las documentaciones y casi idénticos los nombres cuando en realidad el primero lo hace muchas veces y el segundo sólo una vez.

Propongo renombrar el `game.whenCollideDo` a `game.whileCollideRepeat`, y agregar a las respectivas documentaciones la comparación y alguna cosita más que falta:

```
whenCollideDo
Adds a block that will be executed once when the given object collides with other.
Two objects collide when are in the same position.

The block should expect the other object as parameter.

Example:
game.onCollideDo(pepita, { comida => pepita.comer(comida) })

If there are more objects colliding, the block will be executed once for each collider. 

Compare with whileCollideRepeat.
```

```
whileCollideRepeat
Adds a block that will be executed repeatedly while the given object is colliding with other.
Two objects collide when are in the same position.

The block should expect the other object as parameter. Example:
game.whenCollideDo(pepita, { comida => pepita.comer(comida) })

If there are more objects colliding, the block will be executed repeatedly for each collider. 

Compare with onCollideDo.
```

**Comment by fdodino** [comment]
+1, yo también tenía esa confusión. Hay que crear un issue en wollok-language para asociarlo

**Comment by npasserini** [comment]
Es cierto que está confuso, pero no sé si la idea es mantener ambos. Con
rasta pensamos el onCollide porque las múltiples colisiones del tradicional
whenCollide traían problemas. Como yo lo entiendo el otro quedó sólo como
compatibilidad hacia atrás, ¿qué pasa si directamente lo
deprecamos/borramos?

Si lo quieren mantener estoy de acuerdo en cambiar el nombre... aunque
aparece de nuevo el problema de la retrocompatibilidad.

**Comment by npasserini** [comment]
De hecho, si quisiéramos ser prolijos, el plan debería ser primero
deprecarlo y después eliminarlo.
Si lo quieren renombrar, entonces sería crear el nuevo + deprecar y luego
eliminar.

**Comment by asanzo** [comment]
¿Podemos agregar a Wollok warnings de deprecaciones? Podríamos tener una anotación que indique la fecha de deprecación y el mensaje al que depreca, ó algo así. Wollok podría incluso proponer ese refactor. Hmmm esto amerita issue, verdad?

O sea, me gustaría tener soportado en wollok una forma de reconocer y corregir rápidamente esos errores en los proyectos viejos. No sé. Ayer un pibe usó un mensaje para las teclas de game que recibía números ¿game.onKeyPressed ó algo así? y dije "uf, está re mal que se sigan usando cosas deprecadas".

**Comment by npasserini** [comment]
Sí, todo eso se puede, nos vendría bien si nos ayudás a definir qué
deberíamos hacer.

**Comment by asanzo** [comment]
#1843
